packages/runner/src/harness/engine.ts

```diff
@@ -90,16 +90,24 @@ export class Engine extends EventTarget implements Harness {
     source: string,
     options: EngineProcessOptions = {},
   ): Promise<Recipe> {
-    return this.run("/main.tsx", { "/main.tsx": source }, options);
+    return this.run({
+      entry: "/main.tsx",
+      files: [{ name: "/main.tsx", contents: source }]
+    }, options);
   }
 
   async run(
-    entry: string,
-    sources: Record<string, string>,
-    options: EngineProcessOptions,
+    program: Program,
```

Comment:
While we could use `Program` here as an interface, we need to convert it back to a `ProgramResolver` to resolve all of our commontools runtime modules and type environment, which then turns it back into a `Program` (now containing our environment). While `Program` here is simply used as an interface for entry+sources, maybe it'd be more clear if it were not a `js-runtime` type